packages/python/plotly/plotly/express/_core.py

```diff
@@ -1803,31 +2037,38 @@ def process_dataframe_pie(args, trace_patch):
     order_in = args["category_orders"].get(names, {}).copy()
     if not order_in:
         return args, trace_patch
-    df = args["data_frame"]
+    df: nw.DataFrame = args["data_frame"]
     trace_patch["sort"] = False
     trace_patch["direction"] = "clockwise"
-    uniques = list(df[names].unique())
+    uniques = df.get_column(names).unique().to_list()
     order = [x for x in OrderedDict.fromkeys(list(order_in) + uniques) if x in uniques]
-    args["data_frame"] = df.set_index(names).loc[order].reset_index()
+
+    # TODO
+    # Original implementation: args["data_frame"] = df.set_index(names).loc[order].reset_index()
+    # However this is untested, therefore will need some special attention to verify
+    args["data_frame"] = df[order].select(names)
```

Comment:
TODO

Reply:
I've started taking a look at this part @FBruzzesi. It does currently throw an error, but I haven't been able to get it working yet.

Reply:
Sure! Here's the example I've been using

```
import plotly.express as px
import pandas as pd

data = {
    "Category": ["B", "A", "C", "D"],
    "Values": [10, 20, 30, 40]
}

df = pd.DataFrame(data)

category_orders = {"Category": ["C", "B", "A", "D"]}

fig = px.pie(df, 
names="Category", 
values="Values", 
title="Pie Chart with Category Orders",
category_orders=category_orders
)

fig.show()
```

And the error I get its:
```
KeyError: "None of [Index(['C', 'B', 'A', 'D'], dtype='object')] are in the [columns]"
```

Reply:
It should be fixed with the latest commit, however there is no ergonomic way of doing it at the moment. I opened an [issue](https://github.com/narwhals-dev/narwhals/issues/1223), which in my opinion would make it (slightly) better